include/miopen/miopen.h

```diff
@@ -433,13 +439,66 @@ typedef enum {
 /*! @ingroup softmax
  * @enum miopenSoftmaxMode_t
  * Softmax modes
-*/
+ */
 typedef enum {
     MIOPEN_SOFTMAX_MODE_INSTANCE = 0, /*!< compute per image (N) across C, H, W */
     MIOPEN_SOFTMAX_MODE_CHANNEL =
         1, /*!< compute per spatial location (H, W) per image (N) across C */
 } miopenSoftmaxMode_t;
 
+/*! @ingroup TensorReduce
+ * @enum miopenReduceTensorOp_t
+ * Tensor Reduction operation types
+ */
+typedef enum {
+    MIOPEN_REDUCE_TENSOR_ADD = 0, /*!< the operation is adding the values of the reduced elements */
+    MIOPEN_REDUCE_TENSOR_MUL =
+        1, /*!< the operation is multiplying the values of the reduced elements */
+    MIOPEN_REDUCE_TENSOR_MIN =
+        2, /*!< the operation is getting the minimum value of the reduced elements */
+    MIOPEN_REDUCE_TENSOR_MAX =
+        3, /*!< the operation is getting the maximum value of the reduced elements */
+    //MIOPEN_REDUCE_TENSOR_AMAX =
+    //    4, /*!< the operation is getting the maximum absolute value of the reduced elements */
+    //MIOPEN_REDUCE_TENSOR_AVG =
+    //    5, /*!< the operation is getting the averaged value of the reduced elements */
+    //MIOPEN_REDUCE_TENSOR_NORM1 =
+    //    6, /*!< the operation is adding the absolute values of the reduced elements */
+    //MIOPEN_REDUCE_TENSOR_NORM2 = 7, /*!< the operation is getting the square root of the sum of
+    //                                   squares of the reduced elements */
+    //MIOPEN_REDUCE_TENSOR_MUL_NO_ZEROS =
+    //    8, /*!< the operation is same as MUL, but does not have the zero values considered */
+} miopenReduceTensorOp_t;
+
+/*! @ingroup TensorReduce
+ * @enum miopenReduceTensorOp_t
+ * Nan numbers propagation modes
+ */
+typedef enum {
+    MIOPEN_NOT_PROPAGATE_NAN = 0, /*!< does not propagate Nan number */
+    MIOPEN_PROPAGATE_NAN     = 1, /*!< propagate the Nan number by the Reduction operation */
+} miopenNanPropagation_t;
+
+/*! @ingroup TensorReduce
+ * @enum miopenReduceTensorIndices_t
+ * Reduction Indices computation modes
+ */
+typedef enum {
+    MIOPEN_REDUCE_TENSOR_NO_INDICES        = 0, /*!< Does not compuate indices */
+    MIOPEN_REDUCE_TENSOR_FLATTENED_INDICES = 1, /*!< Compute the relative, flatted indices */
+} miopenReduceTensorIndices_t;
```

Comment:
There is no need for an enum. Just add a function like `miopenSetReduceFlattenIndices` to set the mode flatten indices.

Reply:
I don't known whether it is possible to add more types for Indices, so far the only IndicesType is Flatten, so the choice is actually a flag of yes or no

Reply:
One-dimension index even the number of reduced dimensions could be more than one. Refer to cudnnReduceTensor() for the explanation